src/tls/rustls_impl.rs

```diff
@@ -56,6 +58,16 @@ impl TlsHandshaker {
                     OwnedTrustAnchor::from_subject_spki_name_constraints(root.subject, root.spki, root.name_constraints)
                 }));
 
+                #[cfg(feature = "tls-rustls-native-roots")]
+                for native_cert in load_native_certs()? {
+                    let cert = rustls::Certificate(native_cert.0);
+                    // Inspired by https://github.com/seanmonstar/reqwest/blob/231b18f83572836c674404b33cb1ca8b35ca3e36/src/async_impl/client.rs#L363-L365
+                    // Native certificate stores often include certificates with invalid formats,
+                    // but we don't want those invalid entries to invalidate the entire process of
+                    // loading native root certificates
+                    let _ = root_store.add(&cert);
```

Comment:
We could maybe log an error using the `log` crate, this is also done if the proxy environment variables are not set properly.

Reply:
Makes sense, I'll add that :+1: